servers/src/grin/sync/body_sync.rs

```diff
@@ -85,26 +86,42 @@ impl BodySync {
 			return Ok(true);
 		}
 
-		// Find connected peers with strictly greater difficulty than us.
-		let peers_iter = || {
-			self.peers
-				.iter()
-				.with_difficulty(|x| x > head.total_difficulty)
-				.connected()
-		};
+		let peers = {
+			// Find connected peers with strictly greater difficulty than us.
+			let peers_iter = || {
+				// TODO - Once we have "critical mass" of archival nodes on network
```

Comment:
This is commented out currently so we can "bootstrap" a new archive node from nodes known to be archive nodes, but not necessarily advertising as such.
With this change uncommented we would _only_ be able to archival sync (historical blocks) from nodes explicitly advertising via the `BLOCK_HIST` capability flag.

We want to move to this being explicit (capability based) but we can only safely do this if we have enough nodes advertising "archive mode" via the capability flag.

Reply:
This is now uncommented and we filter peers by `BLOCK_HIST` capability when choosing peers to sync blocks from if running in archive mode.